### PR TITLE
Fix printing of external reference addresses in   wasm_application.c

### PR DESCRIPTION
**Pull request by no1wudi**

### CODE REVIEW

File: core/iwasm/common/wasm_application.c [hunk 1]
```diff
@@ -613,7 +613,7 @@ execute_func(WASMModuleInstanceCommon *module_inst, const char *name,
                 u.parts[1] = argv1[k + 1];
                 k += 2;
                 if (u.val && u.val != (uintptr_t)-1LL)
-                    os_printf("0x%" PRIxPTR ":ref.extern", (void *)u.val);
+                    os_printf("%p:ref.extern", (void *)u.val);
```
Comment (yamt) [inline core/iwasm/common/wasm_application.c:616]: u.val is uintptr_t. why don't you remove the cast instead?

File: core/iwasm/common/wasm_application.c [hunk 2]
```diff
@@ -600,7 +600,7 @@ execute_func(WASMModuleInstanceCommon *module_inst, const char *name,
             {
 #if UINTPTR_MAX == UINT32_MAX
                 if (argv1[k] != 0 && argv1[k] != (uint32)-1)
-                    os_printf("0x%" PRIxPTR ":ref.extern", (void *)argv1[k]);
+                    os_printf("%p:ref.extern", (void *)argv1[k]);
```
Comment (yamt) [inline core/iwasm/common/wasm_application.c:603]: iirc, how `%p` converts is implementation-defined.
isn't it a problem for repl?

**Comment by no1wudi** [comment]
@yamt In fact I'm fine to cast them to uintptr_t but it's from stdint.h, but other type such as uint32 is defined internally, mix usage of them looks a little strange.

**Review by yamt — APPROVED** [review]
lgtm

**Pull request merged**